exercises/practice/secret-handshake/src/test/java/HandshakeCalculatorTest.java

```diff
@@ -1,130 +1,100 @@
 import org.junit.Ignore;
-import org.junit.Before;
 import org.junit.Test;
 
-import static java.util.Arrays.asList;
-import static java.util.Collections.emptyList;
-import static java.util.Collections.singletonList;
-import static org.junit.Assert.assertEquals;
+import static org.assertj.core.api.Assertions.assertThat;
 
 public class HandshakeCalculatorTest {
 
-    private HandshakeCalculator handshakeCalculator;
-
-    @Before
-    public void setUp() {
-        handshakeCalculator = new HandshakeCalculator();
-    }
+    private final HandshakeCalculator handshakeCalculator = new HandshakeCalculator();
 
```

Comment:
Since the object has no state, it doesn't matter if we remove the initialization from the `@Before` block?

Reply:
Correct, and even if it did have state, each `@Test` is actually run independently so you would only possibly get an issue if they were running in parallel. Frankly, if a student's solution has any mutable state, that is a bad implementation of this exercise. It may eventually be worth adding a test that does multiple calculations in a row and making sure it always has the correct results across multiple runs (eg. not re-using a mutable list for each calculation).